Move resize / zero logic for _thnn_conv_depthwise2d from codegen to native code.

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #37958 Kill resize-ing and zero-ing from codegen.
* **#37957 Move resize / zero logic for _thnn_conv_depthwise2d from codegen to native code.**
* #37956 Move _thnn_conv2d resize and zero code from codegen to native code.
* #37955 Move resize logic for bmm from codegen to native code.
* #37907 [RESUBMIT] Kill broadcasting from the codegen layer.

Differential Revision: [D21433212](https://our.internmc.facebook.com/intern/diff/D21433212)

gchanan committed:
Move resize / zero logic for _thnn_conv_depthwise2d from codegen to native code.

[ghstack-poisoned]

bhosmer commented on aten/src/ATen/nn_parse.py:
```diff
@@ -295,10 +295,10 @@ def initialize_output_arg(arg):
 
         # grad_weight and grad_bias need to be resized and zeroed
-        if arg['name'] == 'grad_weight' and base['name'] != '_thnn_conv2d':
+        if arg['name'] == 'grad_weight' and base['name'] != '_thnn_conv2d' and base['name'] != '_thnn_conv_depthwise2d':
```
reupping earlier note - personally I'd rather have a list of names that's getting checked here, but not a blocker if you either like it better this way or just want to blast ahead

gchanan replied:
yep, good suggestion.  I knew ahead of time (from checking the generated code) that there were only two cases and I was going to remove them in this stack so it didn't seem worth it.

gchanan committed:
Update on "Move resize / zero logic for _thnn_conv_depthwise2d from codegen to native code."


Differential Revision: [D21433212](https://our.internmc.facebook.com/intern/diff/D21433212)

[ghstack-poisoned]

gchanan committed:
Update on "Move resize / zero logic for _thnn_conv_depthwise2d from codegen to native code."


Differential Revision: [D21433212](https://our.internmc.facebook.com/intern/diff/D21433212)

[ghstack-poisoned]
